### ISSUE TITLE
Characters render on top of ligatures

### ISSUE THREAD

**Issue by stellarskylark** [contour-terminal/contour#1349]
### Contour Terminal version

0.3.12

### Installer source

something else (please specify below)

### Operating System

Fedora Linux 39

### Architecture

x86-64

### Other Software

The Monaspace Xenon font (specifically the Nerd Font)

### Steps to reproduce

Type a ligature, then continue typing, and the new text will overlap the ligature. The cursor position seems to respect the glyphs being multiple columns wide, but the renderer still treats it as a single-width character and draws on top of it. (For the font in question, Monaspace Xenon, you'll need to enable the ligature font features.)

### Actual Behavior

![2023-11-30-142321_1366x768_scrot](https://github.com/contour-terminal/contour/assets/2815515/b688d243-bad2-4944-8aae-24f972b6292e)

### Additional notes

Installed via the official Fedora package

**Comment by stellarskylark** [comment]
Yes, I'm more or less running the default config, minus changing some colors. You can see the full thing here if you think something might be relevant. https://git.solarpunk.moe/stellarskylark/dotfiles/src/branch/main/contour.yml

**Comment by christianparpart** [comment]
Hey @stellarskylark. Many thanks for your interest in Contour.

I was looking into trying to repro this. I however cannot find the font `Monaspace Xenon` as specified by you above on the nerd font github page: https://github.com/ryanoasis/nerd-fonts/releases/latest/

Would you mind directing me to the download link or give me the name of the file I should download to try?

Mind, the `strict_spacing` configuration only mandates whether or not to exclude non-strict-spacing advertised fonts. Experience has (sadly!) shown that it's more often than not a good idea to have that value defaulting to `true`. We should probably change that.

@stellarskylark did you try other fonts also? I am personally using JetBrainsMono (also from nerd fonts), and I see ligatures a lot using that font. :)

Many thanks.

**Comment by stellarskylark** [comment]
@christianparpart Heya yeh, the download link is https://github.com/ryanoasis/nerd-fonts/releases/download/v3.1.1/Monaspace.tar.xz -- it has all five variants in that download, including the slab-serif variant, Xenon. To load it the font name is `MonaspiceXe NF`.

I just tested `JetBrainsMono Nerd Font` and text *does* render correctly for it, so I'll switch to it in the meantime while I wait on a fix for this. (I really like the look of Monaspace, and the "texture healing" feature is just lovely).

I installed Contour via `dnf` to get the official Fedora package. But if you send source code or an RPM I can definitely test it out!
